src/group/libp2p_group_gossip_server.erl

```diff
@@ -415,26 +410,35 @@ schedule_drop_timer(DropTimeOut) ->
     erlang:send_after(DropTimeOut, self(), drop_timeout).
 
 
--spec connections(libp2p_group_gossip:connection_kind() | all, #state{})
-                 -> [{MAddr::string(), Pid::pid()}].
-connections(all, State = #state{workers=Workers}) ->
+-spec connection_pids(libp2p_group_gossip:connection_kind() | all, #{atom() => #{reference() => #worker{}}})
+                 -> [Pid::pid()].
+connection_pids(all, Workers) ->
     maps:fold(
       fun(Kind, _, Acc) ->
-              Conns = connections(Kind, State),
+              Conns = connection_pids(Kind, Workers),
               lists:append(Conns, Acc)
```

Comment:
does order matter here or would consing the new connections onto the accumulator and flattening at the end of the fold also work here?

Reply:
I tried it that way and it didn't seem to help much.